POM/BookStorePage.ts

```diff
@@ -0,0 +1,26 @@
+import { Page } from '@playwright/test';
+
+export class BookStorePage {
+  readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async navigateToBookStore() {
+    await this.page.click('#item-2');
+  }
+
+  async addBookToCollection() {
+    try {
+      await this.page.click('//a[contains(text(),"Git Pocket Guide")]');
+      await this.page.click('#addNewRecordButton');
+      this.page.on('dialog', (dialog) => dialog.accept());
+      await this.page.getByRole('button').click();
+      await this.page.waitForSelector('.text:has-text("Git Pocket Guide")');
+    } catch (error) {
+      console.error('Error adding book to collection:', error);
+      throw error; 
```

Comment:
these selectors are not reusable in this case.
better to store selectors in the constructor and then reuse f.e:
  readonly page: Page;
readonly bookName:Locator
  constructor(page: Page) {
    this.page = page;
this.bookName = this.page.locator('//a[contains(text(),"Git Pocket Guide")]')
  }

then in your test you can use it as:
await this.bookName.click();

Put in click selectors is an old approach